library/ssl_misc.h

```diff
@@ -2313,7 +2313,7 @@ static inline int mbedtls_ssl_tls13_sig_alg_for_cert_verify_is_supported(
     const uint16_t sig_alg)
 {
     switch (sig_alg) {
-#if defined(MBEDTLS_PK_CAN_ECDSA_SOME)
+#if defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED)
```

Comment:
It seems to me that:
- this macro should be guarding the whole function rather than just the ECDSA block (we don't have to worry about `sig_alg_for_cert_verify` if we're not doing ephemeral, as neither PSK nor PSK-ephemeral use a certificate or a signature).
- the ECDSA block still needs to be guarded by something that means "ECDSA is available". We can either keep using `MBEDTLS_PK_CAN` here, or (perhaps a bit better-looking even if equivalent), define in `ssl_ciphersuites.h` a macro named for example `MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_WITH_ECDSA_ENABLED` that is basically `KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED && PK_CAN_ECDSA_SIGN && PK_CAN_ECDSA_VERIFY`.

Reply:
Note: I incorrectly marked this conversation as resolved. This change is incorrect for two reasons:
- the new guard redundant, this whole region of code is already guarded by it (line 2295 above);
- we still need to check if ECDSA is available, so we shoudn't remove `MBEDTLS_PK_CAN_ECDSA_SOME` - unless we replace it with something that somehow also expresses "ECDSA is available".

This was fixed when rebasing #8075 after this PR was merged, see https://github.com/Mbed-TLS/mbedtls/pull/8075#discussion_r1333516744